src/localsocketcontroller.cpp

```diff
@@ -343,9 +351,17 @@ void LocalSocketController::parseCommand(const QByteArray& command) {
   logger.warning() << "Invalid command received:" << command;
 }
 
-void LocalSocketController::write(const QJsonObject& json) {
+void LocalSocketController::write(const QJsonObject& json,
+                                  const QString& expect) {
+  QByteArray payload = QJsonDocument(json).toJson(QJsonDocument::Compact);
+  payload.append('\n');
+
+  if (!expect.isEmpty()) {
+    m_messageExpected = expect;
+    m_messageTimeout.start(CONNECTION_RETRY_INITIAL_MSEC);
```

Comment:
Should this use m_initializingInterval?

Reply:
I don't think so, but it was lazy of me to re-use `CONNECTION_RETRY_INITIAL_MSEC` here. I will add a separate constant for the response timeout.